addons/web_editor/static/src/js/wysiwyg/MoveNodePlugin.js

```diff
@@ -0,0 +1,374 @@
+                        this._currentDropHint.style['left'] = `-${currentDropHintSizeHalf}px`;
+                        this._currentDropHint.style['height'] = `100%`;
+                        this._currentDropHint.style['width'] = `${currentDropHintSize}px`;
+                        dropzoneHintBox.append(this._currentDropHint);
+                        this._currentDropHintElementPosition = ['left', element];
+                    } else if (direction === 'east') {
+                        this._currentDropHint.style['right'] = `-${currentDropHintSizeHalf}px`;
+                        this._currentDropHint.style['height'] = `100%`;
+                        this._currentDropHint.style['width'] = `${currentDropHintSize}px`;
+                        dropzoneHintBox.append(this._currentDropHint);
+                        this._currentDropHintElementPosition = ['right', element];
+                    }
+                });
+                const removeDropHint = () => {
+                    if (this._currentDropHint) {
+                        this._currentDropHint.remove();
+                        this._currentDropHint = null;
+                    }
+                    this._currentDropHintCommand = null;
+                }
+                dropzoneBox.addEventListener('mouseleave', removeDropHint);
+            }
+
+            this._dropzonesContainer.append(dropzoneBox);
+            this._dropzoneHintContainer.append(dropzoneHintBox);
+        }
+    }
+    _stopDropzones(movableElement) {
+        this._editable.classList.remove('oe-editor-dragging');
+        this._dropzonesContainer.replaceChildren();
+        this._dropzoneHintContainer.replaceChildren();
+
+        if (this._currentDropHintElementPosition) {
+            const [position, focusElelement] = this._currentDropHintElementPosition;
+            this._currentDropHintElementPosition = undefined;
+            const previousParent = movableElement.parentElement;
+            if (position === 'top') {
+                focusElelement.before(movableElement);
+            } else if (position === 'bottom') {
+                focusElelement.after(movableElement);
+            }
+            if (previousParent.innerHTML.trim() === '') {
+                const p = document.createElement('p');
+                const br = document.createElement('br');
+                p.append(br);
+                previousParent.append(p);
+            }
+            setSelection(
+                movableElement,
+                movableElement.childNodes.length
+            );
+            resetUnbreakable(movableElement);
+            this._editor.historyStep();
+            // setTimeout as the mutation observer will remove it otherwise.
+            setTimeout(() => this._setMovableElement(movableElement));
+        }
+    }
+    _onMousemove(e) {
+        this._updateAnchorWidgets(e.target);
+    }
+    _onDocumentKeydown() {
+        this._removeMoveWidget && this._removeMoveWidget();
+    }
```

Comment:
I'm confused. What is the use case for document keydown ?

Reply:
comment added